mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1202,29 +1203,17 @@ struct GridwiseGemmV2RewritePattern
     // Logic to setup buffers for blockwise_gemm_v2.
 
     bool isKReduction = (blocksInOutRegs == 1) && (inputSpansPerMfmaIn > 1);
-    int64_t arrayASize = (!isKReduction)
-                             ? (kpacksPerBlock)
-                             : (kpacksPerBlock / inputSpansPerMfmaIn);
-    int64_t arrayBSize = (!isKReduction)
-                             ? (kpacksPerBlock)
-                             : (kpacksPerBlock / inputSpansPerMfmaIn);
+    int64_t inputBufferSize = kpacksPerBlock /
+                              (isKReduction ? inputSpansPerMfmaIn : 1) *
+                              std::max(kpack / k_base, 1L);
 
     Type arrayAType, arrayBType;
     auto privateMemoryAddressSpace = b.getAttr<gpu::AddressSpaceAttr>(
         gpu::GPUDialect::getPrivateAddressSpace());
-    if (kpack > 1) {
-      arrayAType =
-          MemRefType::get({arrayASize}, VectorType::get({kpack}, elementType),
-                          AffineMap{}, privateMemoryAddressSpace);
-      arrayBType =
-          MemRefType::get({arrayBSize}, VectorType::get({kpack}, elementType),
-                          AffineMap{}, privateMemoryAddressSpace);
-    } else {
-      arrayAType = MemRefType::get({arrayASize}, elementType, AffineMap{},
-                                   privateMemoryAddressSpace);
-      arrayBType = MemRefType::get({arrayBSize}, elementType, AffineMap{},
-                                   privateMemoryAddressSpace);
-    }
+    arrayAType = MemRefType::get({inputBufferSize}, mfmaGroup.getArgType(),
+                                 AffineMap{}, privateMemoryAddressSpace);
```

Comment:
I'm a little surprised that you transitioned from a unit of kpack to a unit of k_base. I thought we'd go with unit of 1 and let backend consolidate where it make sense. At the same time, it make sense to pick the unit of k_base because that's eventually what our xdlops will be able to take advantage of.

Reply:
No, yeah, I was remembering why we ended up with buffers of, say, `2xvector<32xf32>` instead of `64xf32` for matrix C.